### PR TITLE
Nextcloud OAuth2

### PR DESCRIPTION
**Pull request by artur-shaik**
related to #322

### CODE REVIEW

File: src/Nextcloud/Provider.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     * {@inheritdoc}
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getInstanceUri().'/index.php/apps/oauth2/authorize', $state);
```
Comment (atymic) [inline src/Nextcloud/Provider.php:29]: Looking at the docs, you shouldn't need the `index.php`?
https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/oauth2.html

File: src/Nextcloud/Provider.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace SocialiteProviders\Nextcloud;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
```
Comment (atymic) [inline src/Nextcloud/Provider.php:8]: Could you add an `@see` docblock with a link to the docs?

File: src/Nextcloud/Provider.php (hunk 1 above)
Comment (artur-shaik) [reply src/Nextcloud/Provider.php:29]: You are right, this is optional. Will fix it later, thanks.

**Review by atymic — APPROVED** [review]
Looks good :)

**Pull request merged**